service/worker/scanner/scanner.go

```diff
@@ -159,10 +159,12 @@ func (s *Scanner) Start() error {
 	}
 
 	var workerTaskQueueNames []string
-	if s.context.cfg.ExecutionsScannerEnabled() {
+	if s.context.cfg.Persistence.DefaultStoreType() != config.StoreTypeSQL && s.context.cfg.ExecutionsScannerEnabled() {
 		s.wg.Add(1)
 		go s.startWorkflowWithRetry(ctx, executionsScannerWFStartOptions, executionsScannerWFTypeName)
 		workerTaskQueueNames = append(workerTaskQueueNames, executionsScannerTaskQueueName)
+	} else if s.context.cfg.ExecutionsScannerEnabled() {
```

Comment:
nit: plz update the [description](https://github.com/temporalio/temporal/blob/44acb3579860b7648fad2e5678619186c6c16f3b/common/dynamicconfig/constants.go#L2612) for this config accordingly. Thanks!